### PR TITLE
Add Cloudflare DNS support

### PR DESCRIPTION
**Pull request by jimkoen**
Hi! This adds support for Cloudflare as a DNS provider.

I think this might be my first, actual, usual contribution to your project! Sorry for opening PR's and then ghosting for months on end, I'm a student though and sometimes don't have time for side projects.

Please review the code. I expect there to be remaining linting errors, which I will fix.
I have have tested my additions on my end, however I would have to look into how to write automated tests for Ansible first to provide additional coverage.

On a sidenote, I'm pretty sure I will be able to finish my PR for Hetzner shortly, I just solved a configuration issue on my end, that kept me from debugging some part of that code.

Regards and happy Pfingsten!

### CODE REVIEW

File: docs/docsite/rst/general_role_parameters.rst [hunk 1]
```diff
@@ -92,11 +92,12 @@ DNS Challenges
 
 If DNS challenges are used, the following variables define how the challenges can be fulfilled:
 
-- ``acme_certificate_dns_provider``: must be one of ``route53``, ``hosttech``, ``ns1``, and ``inwx``. Each needs more information:
+- ``acme_certificate_dns_provider``: must be one of ``route53``, ``hosttech``, ``ns1``, ``inwx`` and ``cloudflare``. Each needs more information:
```
Comment (felixfontein) [inline docs/docsite/rst/general_role_parameters.rst]:
```suggestion
- ``acme_certificate_dns_provider``: must be one of ``route53``, ``hosttech``, ``ns1``, ``inwx``, and ``cloudflare``. Each needs more information:
```
([Oxford comma](https://en.wikipedia.org/wiki/Oxford_comma))

File: roles/acme_certificate/meta/argument_specs.yml [hunk 2]
```diff
@@ -154,6 +155,10 @@ argument_specs:
             In order to use this module you need to disable it.
           - When O(acme_certificate_verify_auth=false), this is not required when O(acme_certificate_dns_provider=inwx) and
             can be provided in other ways, for example with R(module defaults, module_defaults).
+      acme_certificate_cloudflare_token:
+        type: str
+        description:
+          - When using O(acme_certificate_dns_provider=cloudflare), a valid Cloudflare API token must be passed via this option.
```
Comment (felixfontein) [inline roles/acme_certificate/meta/argument_specs.yml:161]:
```suggestion
          - When using O(acme_certificate_dns_provider=cloudflare), a valid Cloudflare API token must be passed via this option.
        version_added: 0.7.0
```

File: roles/acme_certificate/tasks/dns-cloudflare-sanity.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      - acme_certificate_cloudflare_token is defined
+    msg: acme_certificate_cloudflare_token must be specified for Cloudflare
+  run_once: true
+  when: acme_certificate_verify_auth
+
+- name: DNS sanity checks (2/2)
+  ansible.builtin.assert:
+    that:
+      - "'community.general.cloudflare_dns' is community.general.a_module"
+    msg: "community.general.cloudflare_dns must be installed for Cloudflare"
```
Comment (felixfontein) [inline roles/acme_certificate/tasks/dns-cloudflare-sanity.yml]:
```suggestion
    msg: "The community.general collection must be installed for Cloudflare"
```

File: roles/acme_certificate/tasks/dns-cloudflare-create.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+---
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2023, Lucca Jiménez Könings
+
+# Create DNS challenges for DNS provider Cloudflare
+- name: Creating challenge DNS entries for {{ ', '.join(acme_certificate_domains + acme_certificate_ips) }} via Cloudflare API
+  community.general.cloudflare_dns:
+    api_token: "{{ acme_certificate_cloudflare_token }}"
```
Comment (felixfontein) [inline roles/acme_certificate/tasks/dns-cloudflare-create.yml]:
```suggestion
    api_token: "{{ acme_certificate_cloudflare_token | default(omit) }}"
```

File: roles/acme_certificate/tasks/dns-cloudflare-cleanup.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2023, Lucca Jiménez Könings
+
+# Clean up DNS challenges for DNS provider Cloudflare
+- name: Cleaning up challenge DNS entries for {{ ', '.join(acme_certificate_domains + acme_certificate_ips) }} via Cloudflare
+  community.general.cloudflare_dns:
+    api_token: "{{ acme_certificate_cloudflare_token }}"
```
Comment (felixfontein) [inline roles/acme_certificate/tasks/dns-cloudflare-cleanup.yml]:
```suggestion
    api_token: "{{ acme_certificate_cloudflare_token | default(omit) }}"
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments :)

**Comment by jimkoen** [comment]
@felixfontein Does this look okay?

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@jimkoen thanks a lot for your contribution!

**Pull request merged**